src/main2.rs

```diff
@@ -0,0 +1,53 @@
+use cpal::{traits::{DeviceTrait, HostTrait, StreamTrait}, StreamConfig};
+use rustfft::{num_complex::Complex, FftPlanner};
+
+fn main() {
+    let stream = build_stream();
+    loop {
```

Comment:
Two comments about this:

- I think you should only need to start the stream once (so that can happen outside of the while loop).
- Having a busy-loop like this will waste CPU cycles. Instead you can do `loop { std::thread::park(); }` which will suspend the thread forever.